index.js

```diff
@@ -25,15 +25,34 @@ function fastifyCors (fastify, opts, next) {
     allowedHeaders: null,
     maxAge: null,
     preflight: true,
-    hideOptionsRoute: true
+    hideOptionsRoute: true,
+    strictPreflight: false
   }, opts)
 
   const isOriginFalsy = !origin
   const isOriginString = typeof origin === 'string'
   const isOriginFunction = typeof origin === 'function'
 
   if (preflight === true) {
-    fastify.options('*', { schema: { hide: hideOptionsRoute } }, (req, reply) => reply.send())
+    fastify.options('*', { schema: { hide: hideOptionsRoute } }, (req, reply) => {
+      if (origin === false) {
+        reply.send()
```

Comment:
This is how the plugin previously worked.

IMO it would be better:
```javascript
reply.code(404).type('text/html').send('Not Found')
```

But I wanted to keep compatibility :thinking:

Reply:
Go for it.